query_optimizer/OptimizerTree.hpp

```diff
@@ -118,6 +122,169 @@ class OptimizerTree : public OptimizerTreeBase {
   DISALLOW_COPY_AND_ASSIGN(OptimizerTree);
 };
 
+/**
+ * @brief A helper class for printing Protobuf messages in a simple flat-format.
+ * @details Holds key-value pairs of a proto message. Key values are added as strings.
+ *          To use, create a representation and then add properties:
+ *             auto foo = new OptimizerProtoRepresentation<OptimizerPtrType>();
+ *             foo.addProperty("friendly_name", proto.GetExtension(ProtoType::friendly_name));
+ */
+template<class TreeNodeType>
+class OptimizerProtoRepresentation : public OptimizerTreeBase {
```

Comment:
@cramja Hey Marc, I like this idea of helper class to generate string values corresponding to the Protobuf messages. I think this will be useful for when passing the index block properties as well.